param/__init__.py

```diff
@@ -1022,12 +1022,12 @@ def _validate(self, val):
     @classmethod
     def serialize(cls, value):
         if value is None:
-            return 'null'
+            return None
```

Comment:
serialize should not actually return the JSON form but simply a form that can be cleanly serialized. So I don't think this is right.

Reply:
I don't understand your comment. 

As far as I see this change does not return the JSON form, but a form that can be serialized.

Reply:
Sorry I'm just now looking at the handling of None in the serialize/deserialize methods and I believe it to be totally wrong based on the discussion we had in the original serialization PR (see https://github.com/holoviz/param/pull/414).

The `serialize` and `deserialize` methods are meant to return a Python intermediate representation (IR) that should be agnostic to the actual serialization format and it's the responsibility of the `Serializer` class to (de)serialize Python types from/to the underlying format. 'null' is specific to the JSON (and YAML) representation of None, so many serializer won't know what to do with it.

I think most `deserialize` implementations are wrong to check for `value == 'null'` and should instead check for `value is None`.

This being said it might be that it's too late and this should have been caught much earlier.